Assets/BossRoom/Scripts/Shared/Net/GameNetPortal.cs

```diff
@@ -87,18 +87,28 @@ void Start()
 
             NetManager = NetworkManagerGO.GetComponent<NetworkManager>();
 
-            //because we are not a true NetworkedBehavior, we don't get NetworkStart messages. But we still need to run at that point
-            //where we know if we're a host or client. So we fake a "NetworkManager.OnNetworkStarted" event out of the existing OnServerStarted
-            //and OnClientConnectedCallback events.
-            //FIXME_DMW could this be improved?
+            //we synthesize a "NetworkStart" event for the NetworkManager out of existing events. At some point
+            //we expect NetworkManager will expose an event like this itself.
             NetManager.OnServerStarted += NetworkStart;
-            NetManager.OnClientConnectedCallback += (clientId) =>
+            NetManager.OnClientConnectedCallback += ClientNetworkStartWrapper;

```

Comment:
NetworkStart here is confusing. This should fit with the callback it's subscribing to, ie Client Connected or ServerStarted (see above)

Reply:
I agree "NetworkStart" was confusing. I've renamed to "OnNetworkReady", and made the public-facing event "NetworkReadyEvent", which I think more accurately captures the intent (a synonym for OnServerStarted on the server, and OnClientConnected on the client).